From 46d162e8f681498f33d240aba8e410ae82f47ece Mon Sep 17 00:00:00 2001
From: walelm01 <waleed.elmelegy@arm.com>
Date: Fri, 5 Sep 2025 15:45:04 +0100
Subject: [PATCH] Remove AES key wrap Mbedtls APIs

MbedTLS API have transitioned to use PSA key ids
in preparation to be only used through PSA Crypto
APIs and not exposed externally so removing them
to avoid redundancy as we already have PSA Crypto
APIs that can take PSA key ids.

Signed-off-by: walelm01 <waleed.elmelegy@arm.com>
Change-Id: I5d35fd82ff05ea5443e8698950d3063541db6431
---
 .../bootutil/include/bootutil/crypto/aes_kw.h | 33 ++-----------------
 1 file changed, 3 insertions(+), 30 deletions(-)

diff --git a/boot/bootutil/include/bootutil/crypto/aes_kw.h b/boot/bootutil/include/bootutil/crypto/aes_kw.h
index f3c0252f..4cd7a860 100644
--- a/boot/bootutil/include/bootutil/crypto/aes_kw.h
+++ b/boot/bootutil/include/bootutil/crypto/aes_kw.h
@@ -22,15 +22,10 @@
     #error "One crypto backend must be defined: either MBED_TLS or TINYCRYPT"
 #endif
 
-#if defined(MCUBOOT_USE_PSA_CRYPTO)
+#if defined(MCUBOOT_USE_PSA_OR_MBED_TLS)
 #include <psa/crypto.h>
 #endif
 
-#if defined(MCUBOOT_USE_MBED_TLS)
-    #include <mbedtls/aes.h>
-    #include <mbedtls/nist_kw.h>
-#endif /* MCUBOOT_USE_MBED_TLS */
-
 #if defined(MCUBOOT_USE_TINYCRYPT)
     #if defined(MCUBOOT_AES_256)
         #error "Cannot use AES-256 for encryption with Tinycrypt library."
@@ -49,7 +44,7 @@ extern "C" {
 #endif
 
 
-#if defined(MCUBOOT_USE_PSA_CRYPTO)
+#if defined(MCUBOOT_USE_PSA_OR_MBED_TLS)
 
 #define BOOTUTIL_CRYPTO_AES_KW_KEY_SIZE 32
 
@@ -146,29 +141,7 @@ static inline int bootutil_aes_kw_unwrap(bootutil_aes_kw_context *ctx, const uin
     return status;
 }
 
-#elif defined(MCUBOOT_USE_MBED_TLS)
-typedef mbedtls_nist_kw_context bootutil_aes_kw_context;
-static inline void bootutil_aes_kw_init(bootutil_aes_kw_context *ctx)
-{
-    (void)mbedtls_nist_kw_init(ctx);
-}
-
-static inline void bootutil_aes_kw_drop(bootutil_aes_kw_context *ctx)
-{
-    mbedtls_nist_kw_free(ctx);
-}
-
-static inline int bootutil_aes_kw_set_unwrap_key(bootutil_aes_kw_context *ctx, const uint8_t *k, uint32_t klen)
-{
-    return mbedtls_nist_kw_setkey(ctx, MBEDTLS_CIPHER_ID_AES, k, klen * 8, 0);
-}
-
-static inline int bootutil_aes_kw_unwrap(bootutil_aes_kw_context *ctx, const uint8_t *wrapped_key, uint32_t wrapped_key_len, uint8_t *key, uint32_t key_len)
-{
-    size_t olen;
-    return mbedtls_nist_kw_unwrap(ctx, MBEDTLS_KW_MODE_KW, wrapped_key, wrapped_key_len, key, &olen, key_len);
-}
-#endif /* MCUBOOT_USE_MBED_TLS */
+#endif /* MCUBOOT_USE_PSA_OR_MBED_TLS */
 
 #if defined(MCUBOOT_USE_TINYCRYPT)
 typedef struct tc_aes_key_sched_struct bootutil_aes_kw_context;
-- 
2.43.0

